### PR TITLE
Bug fixes and enhancements (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: lib/core/include/apiHandler.hpp [hunk 1]
```diff
@@ -97,235 +100,241 @@ namespace irods {
...
+                irods::log(ERROR(INVALID_ANY_CAST, msg));
+                return INVALID_ANY_CAST;
+            }
+
+            return 0;
+        } // call_handler
+
+        // =-=-=-=-=-=-=-
+        // ctors
+        api_entry( apidef_t& );
```
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp [hunk 2]
```diff
@@ -78,11 +77,15 @@ namespace irods {
...
 
-    template < typename... types_t >
+    template <typename... types_t>
     class api_call_adaptor {
+    private:
         std::function<int(rsComm_t*, types_t...)> fcn_;
+
     public:
-        api_call_adaptor( std::function<int(rsComm_t*, types_t...)> _fcn ): fcn_(_fcn) {
+        api_call_adaptor( std::function<int(rsComm_t*, types_t...)> _fcn )
```
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 1 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 1 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp [hunk 3]
```diff
@@ -97,235 +100,244 @@ namespace irods {
...
+                irods::log(ERROR(INVALID_ANY_CAST, msg));
+                return INVALID_ANY_CAST;
+            }
+
+            return 0;
+        } // call_handler
+
+        // =-=-=-=-=-=-=-
+        // ctors
+        api_entry( apidef_t& );
```
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 3 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

File: lib/core/include/apiHandler.hpp (hunk 2 above)
Comment (trel) [inline lib/core/include/apiHandler.hpp]: Codacy Issue found: [Class 'api_call_adaptor' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)

**Comment by korydraughn** [comment]
Tests passed.  Added #

**Pull request merged**